### PR TITLE
[oneDNN]: Update oneDNN from v2.7.3 to v3.2 on Linux and Windows x86 builds

### PR DESCRIPTION
**Pull request by bhavani-subramanian**
- This PR turns ON oneDNN v3.x by default on both Linux and Windows x86 builds.
- No change is needed from end-users who build TF from source since oneDNN v3.x will be automatically fetched.
- ARM builds will not be affected since they will continue to use oneDNN v2.7.
- This PR also removes conflg settings which was recently added to simultaneously support both oneDNN v2.x and v3.x.
- Support for oneDNN v2.x in TF will be dropped in a future PR.

NOTE: We also need to add `onednn` repo in TSL's workspace [file](https://github.com/google/tsl/blob/main/workspace2.bzl#L157).

### CODE REVIEW

File: tensorflow/tensorflow.bzl [hunk 1]
```diff
@@ -441,9 +440,8 @@ def tf_copts(
         # optimizations for Intel builds using oneDNN if configured
         if_enable_mkl(["-DENABLE_MKL"]) +
         if_mkldnn_openmp(["-DENABLE_ONEDNN_OPENMP"]) +
-        if_onednn_v3(["-DENABLE_ONEDNN_V3"]) +
```
Comment (penpornk) [inline tensorflow/tensorflow.bzl:444]: Please define `onednn_v3_define` in `tsl/mkl/build_defs.bzl` and use it here, e.g., replace
```
if_onednn_v3(["-DENABLE_ONEDNN_V3"]) +
```
with
```
onednn_v3_define() +
```
Where `onednn_v3_define()` returns
```
def onednn_v3_define():
    return select({
        "@org_tensorflow//tensorflow/tsl/mkl:build_with_mkl_aarch64": [],
        "@org_tensorflow//tensorflow/tsl:linux_x86_64": ["-DENABLE_ONEDNN_V3"],
        "@org_tensorflow//tensorflow/tsl:windows": ["-DENABLE_ONEDNN_V3"],
        "//conditions:default": [],
    })
```
(Please also add documentation for `onednn_v3_define`.)

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I think we should aim for minimal changes. Let's keep all `#ifdef ENABLE_ONEDNN_V3` and `ifndef`s the same and just build with `-DENABLE_ONEDNN_V3` for Linux x86 and Windows x86 instead.

File: tensorflow/tensorflow.bzl (hunk 1 above)
Comment (bhavani-subramanian) [reply tensorflow/tensorflow.bzl:444]: Updated.

**Comment by bhavani-subramanian** [comment]
@penpornk Thank you for reviewing this PR. I have addressed your review comments. Please take a look.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Comment by bhavani-subramanian** [comment]
@penpornk Thanks for approving the PR. Please note that we also need to add onednn repo in TSL's workspace [file](https://github.com/google/tsl/blob/main/workspace2.bzl#L157). Would you be able to add it? Thanks.

**Comment by penpornk** [comment]
@bhavani-subramanian Will do. Thank you for the heads up!

Quick notes on CIs for future reference:

[ROCm](http://ml-ci.amd.com:21096/blue/organizations/jenkins/tensorflow%2Fgithub-prs-upstream-master%2FAMD-ROCm-Community-CI-Build/detail/PR-61365/7/pipeline) errors are unrelated. I've seen them on other PRs:
```
ERROR: /workspace/tensorflow/compiler/xla/service/gpu/BUILD:853:11: in deps attribute of cc_library rule //tensorflow/compiler/xla/service/gpu:gpu_executable: Label '//tensorflow/tsl/platform:random' is duplicated

ERROR: /workspace/tensorflow/compiler/xla/service/gpu/BUILD:853:11: Analysis of target '//tensorflow/compiler/xla/service/gpu:gpu_executable' failed
```
[MacOS](https://source.cloud.google.com/results/invocations/5a867fb3-7b77-4530-9041-1ed3a8403069/log) failure is unrelated. (Test timed out)
```
[38,742 / 38,767] 771 / 1022 tests, 1 failed; Testing //tensorflow/python/data/kernel_tests:interleave_test (shard 1 of 24); 18s local, remote-cache ... (16 actions running)
[38,742 / 38,767] 771 / 1022 tests, 1 failed; Testing //tensorflow/python/data/kernel_tests:interleave_test (shard 1 of 24); 27s local, remote-cache ... (16 actions running)


ERROR: Aborting VM command due to timeout of 43200 seconds
```

[Py+CPP Ubuntu CPU](https://source.cloud.google.com/results/invocations/3df54b96-e627-4995-9afa-0ef383d7f6ab/log) failure is unrelated. All 6 failed unit tests timed out.

[Py+CPP Ubuntu GPU](https://source.cloud.google.com/results/invocations/c3b02db3-3730-4e1a-bd80-10e662c6eb89/log) failure is unrelated. The failed test timed out.
```
//tensorflow/python/kernel_tests/linalg:linear_operator_identity_test_gpu TIMEOUT in 2 out of 5 in 466.9s
```

**Pull request merged**